docs/leaderboard/develop/api-reference/index.md

```diff
@@ -0,0 +1,193 @@
+---
+
+import { SdkExampleTabs } from "@site/src/components/SdkExampleTabs";
+// This import is necessary even though it looks like it's un-used; The inject-example-code-snippet
+// plugin will transform instances of SdkExampleTabs to SdkExampleTabsImpl
+import { SdkExampleTabsImpl } from "@site/src/components/SdkExampleTabsImpl";
+
+## Using the Momento Leaderboard
+
+Momento Leaderboard is a first-class service with purpose-built APIs that supports leaderboards with tens of millions of items and rapid ingestion/querying/updates.
+
+## Leaderboard Client
+
+To create and interact with leaderboards, you must first create a LeaderboardClient.
+
+<SdkExampleTabs snippetId={'API_InstantiateLeaderboardClient'} />
+
+Then you can create a Leaderboard by specifying a cache and leaderboard name.
+
+<SdkExampleTabs snippetId={'API_CreateLeaderboard'} />
+
+## Leaderboard Methods
+
+### Upsert elements
+
+Inserts elements if they do not already exist in the leaderboard. Updates elements if they do already exist in the leaderboard.
+
+| Name         | Type                | Description                                                    |
+|--------------|---------------------|----------------------------------------------------------------|
+| elements     | Map<number, number> | A Map or Record of elements (id,score pairs) to upsert.        |
+
+<details>
+  <summary>Method response object</summary>
+
+* Success
+* Error
+
+See [response objects](./response-objects.md) for specific information.
+
+</details>
+
+<SdkExampleTabs snippetId={'API_LeaderboardUpsert'} />
+
+For large leaderboards, you will need to upsert in batches of 8192 elements.
+
+<SdkExampleTabs snippetId={'API_LeaderboardUpsertPagination'} />
+
+### Fetch elements by score
+
+Fetches elements that fall within the specified min and max scores.
+
+
+| Name         | Type                | Description                                                       |
+|--------------|---------------------|-------------------------------------------------------------------|
+| minScore     | Optional[number]    | Inclusive lower bound for the score range. Defaults to `-inf`.    |
+| maxScore     | Optional[number]    | Exclusive upper bound for the score range. Defaults to `+inf`.    |
+| order        | Optional[Ascending / Descending]    | The order to fetch the elements in. Defaults to ascending, meaning 0 is the lowest-scoring rank.   |
+| offset       | Optional[number]    | The number of elements to skip before returning the first element. Defaults to 0. Note: this is not the score of the first element to return, but the number of elements of the result set to skip before returning the first element.    |
+| count        | Optional[number]    | The maximum number of elements to return. Defaults to 8192, which is the maximum that can be fetched per request.    |
+
+<details>
+  <summary>Method response object</summary>
+
+* Success
+    * values(): {`id`: number, `score`: number, `rank`: number}[]
```

Comment:
Do we have an indicator there are more pages if the leaderboard contains more than 8192? Or if it contains more than the provided `count` property?

Reply:
No, the user will have to request more elements until they receive an empty list indicating the end of the results